### PR TITLE
Enable recording of Jest JUnit reports

### LINKED ISSUE #5401: [Jenkinsfile] Allow Jenkins to find our custom JUnit reports
**Is your feature request related to a problem? Please describe.**
Recent work carried out on our testing and Jenkins setup has allowed to output JUnit reports for consumption in Jenkins when running our test suites.

We need to tell Jenkins where the reports are stored to best utilise them.

**Describe the solution you'd like**
Add a post build step to our pipeline pointing to the new JUnit reports.
https://jenkins.io/doc/pipeline/tour/tests-and-artifacts/

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by simonsinclair**
Resolves https://github.com/bbc/simorgh/issues/5401.

~~This PR is currently being debugged here https://github.com/bbc/simorgh/pull/5489.~~

**Overall change:** Add Jest JUnit report recording to Jenkins.

**Code changes:**
- Add `junit` step to Jenkinsfile to record JUnit reports after development tests on `latest`.
  - If report gathering fails, `allowEmptyResults: true` ensures the build will continue.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by karinathomasbbc — APPROVED** [review]

**Comment by chris-hinds** [comment]
What pipelines do you want this to run on? The Jenkins file changed in this PR is used for PR builds.

`Jenkinsfile-e2e` and `Jenkinsfile-e2e-<test|local>` are used for running the full suite of e2e's. Id suspect we mostly want the repirts generated for these pipelines right?

File: Jenkinsfile [hunk 1]
```diff
@@ -283,6 +283,7 @@ pipeline {
       script {
         getCommitInfo()
       }
+      junit 'reports/jest/*.xml'
```
Comment (hindsc52) [inline Jenkinsfile]:
```suggestion
      junit "${env.APP_DIRECTORY}/reports/jest/*.xml"
```

**Comment by simonsinclair** [comment]
> What pipelines do you want this to run on? The Jenkins file changed in this PR is used for PR builds.
> 
> `Jenkinsfile-e2e` and `Jenkinsfile-e2e-<test|local>` are used for running the full suite of e2e's. Id suspect we mostly want the repirts generated for these pipelines right?

This is the report gathering for Jest as opposed to Cypress.
Thanks for the commit suggestion, hopefully that solves the pathing error.

**Comment by simonsinclair** [comment]
De-prioritised. Reports are being generated, however they link to artefacts (such as test suite XML files) that 404 and we don't know why. More investigation is needed at a later stage.

**Pull request closed without merging**